Make the existence of high(type) clearer, especially near subranges (#18286)
nim-lang/Nim: doc/manual.rst

- Ordinal types are countable and ordered. This property allows the operation
  of functions such as `inc`, `ord`, and `dec` on ordinal types to
  be defined.

> - Ordinal values have the smallest possible value, accessible with `low(type)`.

Reviewer: ```suggestion
- Ordinal values have a smallest possible value, accessible with `low(type)`.
```

Reviewer 2: `the` seems more correct here

  Trying to count further down than the smallest value produces a panic or
  a static error.

> - Ordinal values have the largest possible value, accessible with `high(type)`.

Reviewer: ```suggestion
- Ordinal values have a largest possible value, accessible with `high(type)`.
```

Reviewer 2: docs, like code, should be DRY, just say:
```
likewise with `high(type)` for the largest possible value.
```

  type
    Subrange = range[0..5]
    PositiveFloat = range[0.0..Inf]

>     PositiveInt = range[0..high(int)]

Reviewer: Nit: positive means greater than 0, "greater than or equal to 0" would be non-negative.

Reviewer 2: ```suggestion
    Positive* = range[1..high(int)] # as defined in `system`
```
